data/docs/frontend-monitoring/sending-logs.mdx

```diff
@@ -34,139 +145,239 @@ receivers:
         endpoint: 0.0.0.0:4318
         cors:
           allowed_origins:
-            - <<YOUR_FRONTEND_URL>>
+            - <YOUR_FRONTEND_URL>
           allowed_headers: ['*']
 ```
 
-### Step 2: Install dependencies
+### Step 3: Importing the instrumentation file
 
-Install the following dependencies.
+Import the instrumentation file at the top level of your application. This ensures that the OpenTelemetry instrumentation is initialized before any other code runs, allowing it to capture telemetry data from the very beginning of your application's execution.
 
-<Tabs entityName="logs-dependencies">
-  <TabItem value="npm" label="Npm">
-    ```sh
-    npm i \
-      @opentelemetry/resources \
-      @opentelemetry/sdk-logs \
-      @opentelemetry/exporter-logs-otlp-http \
-      @opentelemetry/api-logs
+<Tabs groupId="framework-choice">
+  <TabItem value="react" label="React (Vite/CRA)" default>
+    In your main entry file (typically `main.tsx` or `index.tsx`), import the instrumentation file at the very top:
+
+    ```ts:src/main.tsx
+    import './instrumentation';
+    import { StrictMode } from 'react';
+    import { createRoot } from 'react-dom/client';
+    import './index.css';
+    import App from './App';
+
+    createRoot(document.getElementById('root')!).render(
+      <StrictMode>
+        <App />
+      </StrictMode>
+    );
     ```
+
   </TabItem>
-  <TabItem value="yarn" label="Yarn" default>
-    ```sh
-    yarn add \
-      @opentelemetry/resources \
-      @opentelemetry/sdk-logs \
-      @opentelemetry/exporter-logs-otlp-http \
-      @opentelemetry/api-logs
+  
+  <TabItem value="nextjs-app" label="Next.js (App Router)">
+    Next.js (App Router) supports instrumentation through a special `instrumentation.ts` file at the root of your project (or inside `src/` if using the `src` directory).
+
+    Create `instrumentation.ts` (or `.js`) at the project root:
+
+    ```ts:instrumentation.ts
+    export async function register() {
+      if (process.env.NEXT_RUNTIME === 'nodejs') {
+        // This runs on the server - you can optionally skip client instrumentation here
+        // We'll handle client-side instrumentation separately
+      }
+
+      if (process.env.NEXT_RUNTIME === 'edge') {
+        // This runs on edge runtime
+      }
+    }
+    ```
+
+    For **client-side instrumentation**, create a separate file and import it in your root layout:
+
+    ```ts:app/instrumentation-client.ts
+    // Your OpenTelemetry instrumentation code here
+    // (the code from Step 2)
     ```
+
+    Import it in your root layout:
+
+    ```tsx:app/layout.tsx
+    import './instrumentation-client';
+
+    export default function RootLayout({ children }: { children: React.ReactNode }) {
+      return (
+        <html lang="en">
+          <body>{children}</body>
+        </html>
+      );
+    }
+    ```
+
+    **Note:** Make sure to enable instrumentation in `next.config.js`:
+
+    ```js:next.config.js
+    module.exports = {
```

Comment:
This is not needed in latest nextjs version -  https://signoz-web-git-chore-update-frontend-docs-signoz.vercel.app/docs/instrumentation/opentelemetry-nextjs/